bindings/py/cpp_src/bindings/algorithms/py_HTM.cpp

```diff
@@ -181,6 +188,15 @@ using nupic::algorithms::connections::Permanence;
             self.compute(activeColumns.size(), get_it(activeColumns), learn);
         }, "", py::arg("activeColumns"), py::arg("learn") = true);
 
+        py_HTM.def_property_readonly("connections", [](HTM_t& self)
```

Comment:
please do not expose TM.connections, it should have been protected: in c++ anyway. (I was trying to do that, but lots of tests hijack the functionality)

Reply:
Lots of tests here use that functionality too.
I think it's a legitimate use of the TM, to want to inspect the synapses as these tests do.

Reply:
ok, I now see this is read-only, so then yes, that's fine!